src/App.tsx

```diff
@@ -15,36 +16,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICALLY = 'sort-alphabetically',
+  BY_LENGTH = 'sort-by-length',
+  NONE = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  let visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SortType.BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  const resetSort = () => {
+    setSortField('');
+    setReversed(false);
+  };
```

Comment:
[CODE STYLE] - According to the checklist, function's name should start with a verb. So, instead of `resetSort`, you could name it `handleResetSort` to follow this convention.